lemur/sources/service.py

```diff
@@ -86,84 +86,88 @@ def sync_endpoints(source):
-            continue
-
-        policy = endpoint.pop("policy")
-
-        policy_ciphers = []
-        for nc in policy["ciphers"]:
-            policy_ciphers.append(endpoint_service.get_or_create_cipher(name=nc))
-
-        policy["ciphers"] = policy_ciphers
-        endpoint["policy"] = endpoint_service.get_or_create_policy(**policy)
-        endpoint["source"] = source
-
-        if not exists:
-            current_app.logger.debug(
-                "Endpoint Created: Name: {name}".format(name=endpoint["name"])
-            )
-            endpoint_service.create(**endpoint)
-            new += 1
+                        metrics.send("endpoint.certificate.conflict",
+                                     "gauge", len(lemur_matching_cert),
+                                     metric_tags={"cert": certificate_name, "endpoint": endpoint["name"],
+                                                  "acct": s.get_option("accountNumber", source.options)})
+
+            if not endpoint["certificate"]:
+                current_app.logger.error({
+                    "message": "Certificate Not Found",
+                    "certificate_name": certificate_name,
+                    "endpoint_name": endpoint["name"],
+                    "dns_name": endpoint.get("dnsname"),
+                    "account": s.get_option("accountNumber", source.options),
+                })
+
+                metrics.send("endpoint.certificate.not.found",
+                             "counter", 1,
+                             metric_tags={"cert": certificate_name, "endpoint": endpoint["name"],
+                                          "acct": s.get_option("accountNumber", source.options),
+                                          "dnsname": endpoint.get("dnsname")})
+                continue
+
+            policy = endpoint.pop("policy")
+
+            policy_ciphers = []
+            for nc in policy["ciphers"]:
+                policy_ciphers.append(endpoint_service.get_or_create_cipher(name=nc))
+
+            policy["ciphers"] = policy_ciphers
+            endpoint["policy"] = endpoint_service.get_or_create_policy(**policy)
+            endpoint["source"] = source
+
+            if not exists:
+                current_app.logger.debug(
+                    "Endpoint Created: Name: {name}".format(name=endpoint["name"])
+                )
+                endpoint_service.create(**endpoint)
+                new += 1
 
-        else:
-            current_app.logger.debug("Endpoint Updated: {}".format(endpoint))
-            endpoint_service.update(exists.id, **endpoint)
-            updated += 1
+            else:
+                current_app.logger.debug("Endpoint Updated: {}".format(endpoint))
+                endpoint_service.update(exists.id, **endpoint)
+                updated += 1
+        except Exception as e:
+            current_app.logger.warning(f"Error in processing sync endpoint "
```

Comment:
We use `capture_exception()` in many (but not all) situations like this; seems like we may want it here?